FAIMS/FAIMS3
# Improve Auto-incrementers

# Fix: Improve Auto-incrementers

## JIRA Ticket

None

## Description

Auto increment fields are used to generate human readable IDs for records.  There 
was a bug that caused them to not work if the initial value of the field was 
set to "" as it was in the designer. 

These fields require that a range of numbers is set for the device before records can
be properly created.  The past behaviour was to notice when they weren't set up but
not really help the user set them.

## Proposed Changes

- Deals with the case when the initial value is ""
- Turn the range setting form into a popup dialog
- trigger the popup if a user tries to create a record without the ranges being set

## How to Test

Open a notebook with an auto increment field configured, eg. [Campus Survey](https://github.com/FAIMS/FAIMS3-Notebook-Campus-Survey-Demo).  Try to create
a record after activating the notebook, you should see a popup to allow you to set
the ranges.

Same popup is accessible via the Settings tab on the notebook.


## Checklist
